### ISSUE TITLE
allow streamed output writing to parquet/csv rather than aggregating predictions/embeddings in memory

### ISSUE THREAD

**Issue by sammlapp** [kitzeslab/opensoundscape#1277]
often don't need the predictions in memory, and they get large (we work around this by dividing inference tasks into subsets)

model.predict(logits_dest='./logits.parquet', embeddings_dest='embeddings.csv',return_scores=False)

-> options to save separate files per file/folder
-> toggle whether to aggregate scores in memory or only write to file
-> support .parquet and .csv
